Mathlib/Data/SetLike/Closure.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2025 Artie Khovanov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artie Khovanov
+-/
+import Mathlib.Data.SetLike.Basic
+import Mathlib.Data.Set.Lattice
+
+/-!
+# Typeclass for lattices with a closure-adjoint embedding into a power set.
+
+This file defines a typeclass for lattices `L` that embed nicely into a power set `Set α`,
+in the sense that the closure operation `Set α → L` is adjoint to the embedding. The main purpose
+of this class is to abstract the closure construction common to most algebraic substructures.
+
+
+## Main definitions
+
+* `SetLikeCompleteLattice` : class for complete lattices that have an embedding into some `Set α`
+  which preserves the order and arbitrary infima.
+* `SetLike.closure` : the natural closure operation on `Set α` with respect to a
+  `SetLikeCompleteLattice` embedding; that is, the map from a set to the least lattice element that
+  contains it.
+
+## Main result
+
+* `SetLike.gi_closure` : for a `SetLikeCompleteLattice`, the operations
+  `SetLike.closure` and `SetLike.coe` (the embedding) form a Galois insertion.
+
+-/
+
+class SetLikeCompleteLattice (L α : Type*) extends CompleteLattice L, SetLike L α where
+  le_def' : le =
+    @LE.le L (@Preorder.toLE L (@PartialOrder.toPreorder L SetLike.instPartialOrder))
+  lt_def' : lt =
```

Comment:
In any preorder we have `x < y ↔ x ≤ y ∧ ¬ y ≤ x`, so this should follow from `le_def'`.